### PR TITLE
Validate request protocol

### PR DESCRIPTION
**Pull request by JamesNK**
Fixes https://github.com/grpc/grpc-dotnet/issues/474

### CODE REVIEW

File: src/Grpc.AspNetCore.Server/Internal/CallHandlers/ServerCallHandlerBase.cs [hunk 1]
```diff
@@ -55,7 +56,14 @@ public Task HandleCallAsync(HttpContext httpContext)
...
             {
-                GrpcProtocolHelpers.SendHttpError(httpContext.Response, StatusCodes.Status415UnsupportedMediaType, StatusCode.Internal, error!);
+                GrpcProtocolHelpers.SendHttpError(httpContext.Response, StatusCodes.Status415UnsupportedMediaType, StatusCode.Internal, error);
+                return Task.CompletedTask;
+            }
+            if (httpContext.Request.Protocol != GrpcProtocolConstants.Http2Protocol)
+            {
+                var protocolError = $"Request protocol '{httpContext.Request.Protocol}' is not supported.";
+                GrpcProtocolHelpers.SendHttpError(httpContext.Response, StatusCodes.Status426UpgradeRequired, StatusCode.Internal, protocolError);
+                httpContext.Response.Headers[HeaderNames.Upgrade] = GrpcProtocolConstants.Http2Protocol;
```
Comment (JamesNK) [inline src/Grpc.AspNetCore.Server/Internal/CallHandlers/ServerCallHandlerBase.cs]: @Tratcher @halter73 Is 426 the right status to return when a HTTP/1.1 request comes to a gRPC?

Comment (Tratcher) [reply src/Grpc.AspNetCore.Server/Internal/CallHandlers/ServerCallHandlerBase.cs]: Anything you put down will be primarily for diagnostic purposes, there's no automatic transition support beyond ALPN.
426 is OK, 421 looks like the second best option.

Comment (JamesNK) [reply src/Grpc.AspNetCore.Server/Internal/CallHandlers/ServerCallHandlerBase.cs]: Diagnostics and providing a clear error message is the goal here. Currently HTTP/1.1 requests get through to the gRPC request delegate and only fail when the framework tries to get or set trailers.

Will use 426.

File: test/FunctionalTests/Client/CompressionTests.cs [hunk 2]
```diff
@@ -41,7 +41,7 @@ public async Task SendCompressedMessage_ServiceCompressionConfigured_ResponseGzi
 
             string? requestMessageEncoding = null;
             string? responseMessageEncoding = null;
-            using var httpClient = Fixture.CreateClient(new TestDelegateHandler(
+            using var httpClient = Fixture.CreateClient(messageHandler: new TestDelegateHandler(
```
Comment (davidfowl) [inline test/FunctionalTests/Client/CompressionTests.cs]: ?

**Pull request merged**